servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/H2PriorKnowledgeFeatureParityTest.java

```diff
@@ -435,6 +442,126 @@ public void serverHeaderCookieRemovalAndIteration() throws Exception {
         }
     }
 
+    @Test
+    public void clientSendsLargerContentLength() throws Exception {
+        assumeTrue(h2PriorKnowledge); // http/1.x will timeout waiting for more payload.
+        clientSendsInvalidContentLength(1, false);
+    }
+
+    @Test
+    public void clientSendsLargerContentLengthTrailers() throws Exception {
+        assumeTrue(h2PriorKnowledge); // http/1.x will timeout waiting for more payload.
+        clientSendsInvalidContentLength(1, true);
+    }
+
+    @Test
+    public void clientSendsSmallerContentLength() throws Exception {
+        clientSendsInvalidContentLength(-1, false);
+    }
+
+    @Test
+    public void clientSendsSmallerContentLengthTrailers() throws Exception {
+        clientSendsInvalidContentLength(-1, true);
+    }
+
+    private void clientSendsInvalidContentLength(int contentLengthAdder, boolean addTrailers) throws Exception {
+        InetSocketAddress serverAddress = bindHttpEchoServer();
+        try (BlockingHttpClient client = forSingleAddress(HostAndPort.of(serverAddress))
+                .protocols(h2PriorKnowledge ? h2Default() : h1Default())
+                .executionStrategy(clientExecutionStrategy)
+                .appendClientFilter(client1 -> new StreamingHttpClientFilter(client1) {
+                    @Override
+                    protected Single<StreamingHttpResponse> request(final StreamingHttpRequester delegate,
+                                                                    final HttpExecutionStrategy strategy,
+                                                                    final StreamingHttpRequest request) {
+                        return request.toRequest().map(req -> {
+                            req.headers().remove(TRANSFER_ENCODING);
+                            req.headers().set(CONTENT_LENGTH,
+                                    String.valueOf(req.payloadBody().readableBytes() + contentLengthAdder));
+                            return req.toStreamingRequest();
+                        }).flatMap(req -> delegate.request(strategy, req));
+                    }
+                }).buildBlocking()) {
+            HttpRequest request = client.get("/").payloadBody("a", textSerializer());
+            if (addTrailers) {
+                request.trailers().set("mytrailer", "myvalue");
+            }
+            if (h2PriorKnowledge) {
+                assertThrows(Http2Exception.H2StreamResetException.class, () -> client.request(request));
+            } else {
+                ReservedBlockingHttpConnection reservedConn = client.reserveConnection(request);
+                try {
+                    reservedConn.request(request);
+                    assertThrows(ClosedChannelException.class, () ->
+                            reservedConn.request(client.get("/").payloadBody("a", textSerializer())));
```

Comment:
You already have prepared `request`, consider using that one

Reply:
I didn't want to re-use bcz the request has a payload body. I'll remove the payload body from the second request bcz it isn't necessary.